apps/src/lib/levelbuilder/course-editor/CourseEditor.js

```diff
@@ -57,44 +60,107 @@ class CourseEditor extends Component {
+      familyName: this.props.initialFamilyName,
+      versionYear: this.props.initialVersionYear,
+      isStable: this.props.initialIsStable,
+      scriptsInCourse: this.props.initialScriptsInCourse
     };
   }
 
   handleUpdateAnnouncements = newAnnouncements => {
     this.setState({announcements: newAnnouncements});
   };
 
+  handleSave = (event, shouldCloseAfterSave) => {
+    event.preventDefault();
+
+    this.setState({isSaving: true, lastSaved: null, error: null});
+
+    let dataToSave = {
+      title: this.state.title,
+      version_title: this.state.versionTitle,
+      description_short: this.state.descriptionShort,
+      description_student: this.state.descriptionStudent,
+      description_teacher: this.state.descriptionTeacher,
+      has_verified_resources: this.state.hasVerifiedResources,
+      has_numbered_units: this.state.hasNumberedUnits,
+      family_name: this.state.familyName,
+      version_year: this.state.versionYear,
+      is_stable: this.state.isStable,
+      visible: this.state.visible,
+      pilot_experiment: this.state.pilotExperiment,
+      scripts: this.state.scriptsInCourse
+    };
+
+    if (this.props.migratedTeacherResources) {
+      dataToSave.resourceIds = this.props.migratedTeacherResources.map(
+        r => r.id
+      );
+    }
+
+    if (this.props.studentResources) {
+      dataToSave.studentResourceIds = this.props.studentResources.map(
+        r => r.id
+      );
+    }
+
+    $.ajax({
+      url: `/courses/${this.props.name}`,
+      method: 'PUT',
+      dataType: 'json',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify(dataToSave)
+    })
+      .done(data => {
+        if (shouldCloseAfterSave) {
+          navigateToHref(linkWithQueryParams(data.coursePath));
+        } else {
+          this.setState({
+            lastSaved: Date.now(),
+            isSaving: false
+          });
+        }
+      })
+      .fail(error => {
+        this.setState({isSaving: false, error: error.responseText});
+      });
+  };
+
   render() {
+    const {name, scriptNames, courseFamilies, versionYearOptions} = this.props;
     const {
-      name,
+      announcements,
+      teacherResources,
       title,
       versionTitle,
+      descriptionShort,
+      hasVerifiedResources,
+      hasNumberedUnits,
       familyName,
       versionYear,
-      descriptionShort,
-      scriptsInCourse,
-      scriptNames,
-      courseFamilies,
-      versionYearOptions
-    } = this.props;
-    const {announcements, teacherResources} = this.state;
+      isStable,
+      scriptsInCourse
+    } = this.state;
     return (
       <div>
         <h1>{name}</h1>
         <label>
           Display Name
-          <input
-            type="text"
-            name="title"
-            defaultValue={title}
-            style={styles.input}
-          />
+          <input type="text" defaultValue={title} style={styles.input} />
```

Comment:
I think you might need to have an `onChange` handler here. When I update the display name locally, the change doesn't stick

Reply:
Also for version year and short description

Reply:
Oh good catch. I did not add that. Thanks for your thoughtful review